### PR TITLE
Factorize code with ErrorsWrapper and Utils.nullToEmpty

### PR DESCRIPTION
**Pull request by Gawen-pjr**
Just propagated usage of ErrorsWrapper interface and Utils.nullToEmpty static method for all domain classes.

### CODE REVIEW

**Comment by cdancy** [comment]
Have you run the mockTests locally (`./gradlew clean build mockTest`)? You should see failures.

File: src/main/java/com/cdancy/bitbucket/rest/domain/branch/Branch.java [hunk 1]
```diff
@@ -45,15 +47,12 @@
...
-
     Branch() {
     }
 
     @SerializedNames({ "id", "displayId", "type", "latestCommit", "latestChangeset", "isDefault", "errors" })
     public static Branch create(String id, String displayId, String type,
                                 String latestCommit, String latestChangeset, boolean isDefault, List<Error> errors) {
-        return new AutoValue_Branch(id, displayId, type, latestCommit, latestChangeset, isDefault,
-                errors != null ? ImmutableList.copyOf(errors) : ImmutableList.<Error> of());
+        return new AutoValue_Branch(Utils.nullToEmpty(errors), id, displayId, type, latestCommit, latestChangeset, isDefault);
```
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/domain/branch/Branch.java]: The `Utils.nullToEmpty(errors)` should be the last arg.

Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/domain/branch/Branch.java]: Or maybe not if your implementing from super class?

**Comment by Gawen-pjr** [comment]
All MockTests pass when I build locally. Travis also seems happy.

File: src/main/java/com/cdancy/bitbucket/rest/domain/branch/Branch.java (hunk 1 above)
Comment (Gawen-pjr) [inline src/main/java/com/cdancy/bitbucket/rest/domain/branch/Branch.java]: AutoValue generated class imposes me to pass errors as first argument.

File: src/main/java/com/cdancy/bitbucket/rest/domain/common/Utils.java [hunk 2]
```diff
@@ -18,11 +18,18 @@
 package com.cdancy.bitbucket.rest.domain.common;
 
 import java.util.List;
+import java.util.Map;
 
 import com.google.common.collect.ImmutableList;
+import com.google.common.collect.ImmutableMap;
 
 public class Utils {
```
Comment (Gawen-pjr) [inline src/main/java/com/cdancy/bitbucket/rest/domain/common/Utils.java]: Yes I agree

**Comment by cdancy** [comment]
@Gawen-pjr disregard that comment. It was in response to where you placed the `Utils.toNull` in the AutoValue constructor.

**Comment by cdancy** [comment]
@Gawen-pjr very awesome and glad you tackled this next. I was hoping your next PR was going to be this otherwise I was going to hop in and finish the work you started ;)

File: src/main/java/com/cdancy/bitbucket/rest/domain/branch/Branch.java (hunk 1 above)
Comment (cdancy) [inline src/main/java/com/cdancy/bitbucket/rest/domain/branch/Branch.java]: Fair enough and wasn't aware of that.

**Comment by Gawen-pjr** [comment]
I'm currently factorizing classes with "Links" field which is also widely used. If you can wait 10 minutes before merging, I'll push the changes.

**Comment by cdancy** [comment]
@Gawen-pjr still reviewing on this end. I'll stop until I have the go-ahead from you that you're finished.

**Comment by cdancy** [comment]
@Gawen-pjr actually ... is you're looking to factor out other variables best to send that in a separate PR.

**Comment by cdancy** [comment]
@Gawen-pjr I like this PullRequest as-is. Lets not add anything else unless it pertains to these changes.

Thoughts?

**Comment by cdancy** [comment]
@Gawen-pjr we can do these but no more in this PR!!! ;) I'll allow it because it cleans things up but I generally don't like to review large PR's. They should be as small as possible and address a single issue.

**Comment by Gawen-pjr** [comment]
Aw, just pushed last changes. I'll rollback with a reverse commit

**Comment by Gawen-pjr** [comment]
OK. I leave this PR as is. Starting a new PR for Links factorization. Sorry for the disturbances.

**Comment by cdancy** [comment]
Merged. Thanks for the code-cleanup/PR @Gawen-pjr !!!

**Comment by Gawen-pjr** [comment]
OK. See you in next PR ;-)

**Pull request merged**